contracts/TokenFactory.sol

```diff
@@ -263,31 +312,59 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         return (_amount * _feePercent) / FEE_DENOMINATOR;
     }
 
-    function setWinner(address winnerAddress) external onlyOwner {
-        winnerToken = winnerAddress;
+    function getWinnerByDay(uint256 day) public view returns (address) {
+        uint256 maxDayCollateral = 0;
+        address winnerAddress;
+
+        for (uint256 i = 0; i < totalTokensAddresses; i++) {
+            address tokenAddress = tokensAddresses[i];
+            uint256 _collateral = collateralByDay[day][tokenAddress];
+
+            if (_collateral > maxDayCollateral) {
+                maxDayCollateral = _collateral;
+                winnerAddress = tokenAddress;
+            }
+        }
+
+        return winnerAddress;
+    }
+
+    function setWinner() external {
```

Comment:
Since this function is permissionless, it could accept a parameter "competition id", instead of hard-coded to be only working for setting previous day's competition